test/libp2p/crypto/rsa_provider_test.cpp

```diff
@@ -0,0 +1,250 @@
+      0x39, 0x7b, 0x14, 0xa3, 0xdc, 0xbd, 0x01, 0xea, 0x72, 0xdf, 0xc0, 0x3e, \
+      0x9f, 0xc0, 0xe3, 0x22, 0x91, 0xcb, 0xa1, 0x5b, 0xe7, 0x46, 0x22, 0x71, \
+      0x59, 0x59, 0xa6, 0x20, 0xdf, 0x43, 0x12, 0xcf, 0x1e, 0x8b, 0x1d, 0x3d, \
+      0xfa, 0x50, 0xa9, 0x98, 0x8b, 0x80, 0x7b, 0x49, 0x6d, 0xfd, 0x85, 0xe2, \
+      0x7c, 0x01, 0xea, 0x52, 0xbb, 0xee, 0x8c, 0xab, 0x9a, 0xd5, 0x96, 0xe0, \
+      0xca, 0x73, 0xd6, 0xee, 0x52, 0x20, 0xb9, 0x6e, 0x0d, 0x5c, 0xa8, 0x3e, \
+      0x8d, 0x19, 0xfc, 0xb1, 0x69, 0x27, 0x17, 0xc4, 0x73, 0xa6, 0xee, 0x39, \
+      0xc5, 0x76, 0x31, 0xc7, 0x55, 0x28, 0xd0, 0xcb, 0x2a, 0x88, 0xeb, 0x55, \
+      0x34, 0xbe, 0x97, 0xdd, 0x6e, 0xfc, 0xed, 0x61, 0x9c, 0x26, 0x01, 0x78, \
+      0xd8, 0x3c, 0x1c, 0xb5, 0x4b, 0xe9, 0x1c, 0x15, 0x50, 0x51, 0x52, 0xbb, \
+      0x57, 0x4e, 0x37, 0x44, 0x1c, 0xa0, 0xa6, 0x8b, 0x4a, 0xd6, 0x0c, 0x8d, \
+      0x76, 0x40, 0x72, 0xcf, 0xfc, 0x16, 0x29, 0xb3, 0x56, 0x40, 0xab, 0x84, \
+      0x37, 0x79, 0x2f, 0x95, 0x0e, 0x3c, 0x17, 0x0b, 0xe7, 0xcd, 0x13, 0x64, \
+      0x1d, 0xd0, 0xfe, 0x66, 0x44, 0x24, 0x6e, 0x0b, 0x6a, 0x9d, 0x03, 0x7d, \
+      0xf4, 0x3b, 0xd1, 0x40, 0xda, 0xc0, 0x93, 0x97, 0x6a, 0x72, 0xda, 0xfb, \
+      0xb7, 0xe9, 0xd6, 0x51
+
+class RsaProviderTest : public ::testing::Test {
+ protected:
+  RsaProviderImpl provider_;
+  Key private_key_{SAMPLE_RSA_PRIVATE_KEY_BYTES};
+  Key public_key_{SAMPLE_RSA_PUBLIC_KEY_BYTES};
+  std::vector<uint8_t> message_{SAMPLE_MESSAGE_BYTES};
+  std::vector<uint8_t> signature_{SAMPLE_RSA_SIGNATURE_BYTES};
+};
+
+/**
+ * @given Pre-generated RSA public key and signature using reference golang
+ *        implementation
+ * @when Verifying RSA signature
+ * @then Pre-generated signature must be valid
+ */
+TEST_F(RsaProviderTest, SignatureVerificationSuccess) {
+  EXPECT_OUTCOME_TRUE(result,
+                      provider_.Verify(message_, signature_, public_key_));
+  ASSERT_TRUE(result);
+}
+
+/**
+ * @given Pre-generated RSA public key and invalid RSA signature
+ * @when Verifying RSA signature
+ * @then Signature verification must be successful with result "Invalid
+ * signature"
+ */
+TEST_F(RsaProviderTest, InvalidSignatureVerificationFailure) {
+  Signature different_signature{SAMPLE_RSA_INVALID_SIGNATURE_BYTES};
+  EXPECT_OUTCOME_TRUE(
+      result, provider_.Verify(message_, different_signature, public_key_));
+  ASSERT_FALSE(result);
+}
+
+/**
+ * @given Pre-generated RSA signature and invalid public key
+ * @when Verifying RSA signature
+ * @then Signature verification must be successful with result "Invalid
+ * signature"
+ */
+TEST_F(RsaProviderTest, InvalidPublicKeySignatureVerificationFailure) {
+  Key invalid_key = public_key_;
+  invalid_key[32] ^= 8;
+  EXPECT_OUTCOME_TRUE(result,
+                      provider_.Verify(message_, signature_, invalid_key));
+  ASSERT_FALSE(result);
+}
+
+TEST_F(RsaProviderTest, SigningSample) {
```

Comment:
no comments. Also looks like we need to check that signature was correct using verify